### PR TITLE
[i2c, test] Add i2c_host_accelerometer_test

### PR DESCRIPTION
**Pull request by jwnrt**
This PR adds an i2c host test for using an accelerometer device.

Writes and reads back the accelerometer's "THRESH_TAP" (tap detection threshold) register, then enables and reads measurements.

Can't check whether the measurements are correct, so just checks they're not all zero (which is possible, but more likely shows an error).

### CODE REVIEW

File: sw/device/tests/BUILD
Comment (timothytrippel) [inline sw/device/tests/BUILD:1767]: The "skip_in_ci" tag is a bit of a hack used to not run all ROM E2E tests that had many different permutations since they used different OTP images with different LC states. However they do get run in a nightly round.

If you want this to be run only with special HW hooked up, then maybe you should just tag this test with "manual" instead. Unless there is a specific CI machine in CI (i.e. Bob?) that can run this?

**Review by timothytrippel — APPROVED** [review]

File: sw/device/tests/BUILD
Comment (jwnrt) [reply sw/device/tests/BUILD:1767]: @engdoreis @johngt do we want to run these in CI? Every board has a BoB connected.

Comment (johngt) [reply sw/device/tests/BUILD:1767]: The intention is to run these tests in CI. All boards in CI, current and planned have a BoB either attached or ready to be attached to them.

**Review by engdoreis — APPROVED** [review]

File: sw/device/tests/BUILD
Comment (engdoreis) [reply sw/device/tests/BUILD:1767]: We will have the bitstream that enables the BoB in CI very soon, so I'll merge this PR and after the bitstream PR gets merged I'll file a PR removing this tag.

**Pull request merged**
